dm/syncer/err-operator/operator.go

```diff
@@ -94,6 +103,59 @@ func (h *Holder) Set(pos string, op pb.ErrorOp, events []*replication.BinlogEven
 	return nil
 }
 
+// GetBehindCommands gets behind commands.
+func (h *Holder) GetBehindCommands(pos string) []pb.HandleWorkerErrorRequest {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	current := mysql.Position{}
+	fmt.Sscanf(strings.ReplaceAll(pos[1:len(pos)-1], ",", ""), "%s %d", &current.Name, &current.Pos)
+
+	// find behind position
+	var behindPositions []mysql.Position
+	for key := range h.operators {
+		p := mysql.Position{}
+		fmt.Sscanf(strings.ReplaceAll(key[1:len(key)-1], ",", ""), "%s %d", &p.Name, &p.Pos)
+		if current.Compare(p) <= 0 {
+			behindPositions = append(behindPositions, p)
+		}
+	}
+	sort.Slice(behindPositions, func(i, j int) bool {
+		return behindPositions[i].Compare(behindPositions[j]) < 0
+	})
+
+	res := make([]pb.HandleWorkerErrorRequest, 0, len(behindPositions))
+	for _, behindPosition := range behindPositions {
+		res = append(res, h.operators[behindPosition.String()].originReq)
+	}
+
+	return res
+}
+
+func (h *Holder) SetHasAllInjected(startLocation binlog.Location) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
```

Comment:
ditto

Reply:
`SetHasAllInjected` will set `isAllInjected` be true. so `Lock` may be better.